streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -487,7 +508,11 @@ public void advanceWallClockTime(final long advanceMs) {
      * @see #getSessionStore(String)
      */
     public StateStore getStateStore(final String name) {
-        return ((ProcessorContextImpl) task.context()).getStateMgr().getStore(name);
+        StateStore res = task == null ? null : 
+            ((ProcessorContextImpl) task.context()).getStateMgr().getStore(name);
+        if (res == null && globalStateManager != null)
+            res = globalStateManager.getGlobalStore(name);
+        return res;
```

Comment:
This is a pretty small method, so it might not matter that much, but sequences of assignment and matching can become a little confusing. What do you think about this instead:
```java
if (task != null) {
    return ((ProcessorContextImpl) task.context()).getStateMgr().getStore(name);
} else if (globalStateManager != null) {
    return globalStateManager.getGlobalStore(name);
} else {
    return null;
}
```

Reply:
The code you propose doesn't do the same thing, I need to check the presence of a global store event in the task exists but returns null (ie. there is no local store with that name).

Reply:
Ah, I see. In that case, I would advocate for something like the following:
```java
final StateStore localStore = task == null ? null : ((ProcessorContextImpl) task.context()).getStateMgr().getStore(name);
if (localStore != null) {
    return localStore;
} else {
    return globalStateManager == null ? null : globalStateManager.getGlobalStore(name);
}
```